File: backend/auth/src/main/java/auth/pymes/service/InvitationService.java
package auth.pymes.service;

import auth.pymes.common.models.dto.request.CreateInvitationRequest;
import auth.pymes.common.models.dto.request.InvitationRegisterRequest;
import auth.pymes.common.models.dto.response.AuthResponse;
import auth.pymes.common.models.dto.response.InvitationInfoResponse;
import auth.pymes.common.models.dto.response.InvitationResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface InvitationService {

    InvitationInfoResponse getInvitationInfo(String token);

    AuthResponse registerAndAccept(String token, InvitationRegisterRequest request);

    Page<InvitationResponse> getPendingInvitations(Pageable pageable, Object principal);

    InvitationResponse createInvitation(CreateInvitationRequest request, Object principal);

    InvitationResponse acceptInvitation(String invitationToken, Object principal);

    void cancelInvitation(UUID invitationId, Object principal);
}

File: frontend/pymes/src/modules/auth/services/invitation.service.ts
import { api } from 'src/boot/axios';
import type { InvitationRegisterRequest } from '../types';

export const invitationService = {
  getInvitationInfo(token: string) {
    return api.get(`/invitations/${token}/info`);
  },

  registerAndAccept(token: string, data: InvitationRegisterRequest) {
    return api.post(`/invitations/${token}/register`, data);
  },

  getPendingInvitations(page = 0, size = 10) {
    return api.get('/invitations', {
      params: { page, size }
    });
  },

  createInvitation(data: { tenantId: string; email: string; role: string }) {
    return api.post('/invitations', data);
  },

  cancelInvitation(invitationId: string) {
    return api.delete(`/invitations/${invitationId}`);
  },

  acceptInvitation(invitationToken: string) {
    return api.post('/invitations/accept', { invitationToken });
  }
};